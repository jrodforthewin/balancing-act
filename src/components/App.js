
import React, {useRef, useState} from 'react';
import logo from './logo.svg';
import './App.css';
import isBalanced from '../algorithms/balancer';

function App() {
  const textRef = useRef("");
  const [balancedResponse, setBalanced] = useState("");

  const submit = (e) => {
    e.preventDefault();
    const textVal = textRef.current.value;
    setBalanced(isBalanced(textVal));
  }

  return (
    <div className="App">
      <header className="App-header">
      <form onSubmit={submit}>
       <input ref={textRef} type="text" onChange={submit}></input>
       <button>Send Ref</button>
       </form>
       <h3>My app</h3>
       <div>{balancedResponse? "Yes" : "No"}</div>
       Learn react
      </header>
    </div>
  );
}

export default App;
