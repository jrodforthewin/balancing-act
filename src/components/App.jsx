import React, { useState } from 'react';
import './App.css';
import {
  Container,
  Box,
  Typography,
  TextField,
} from '@material-ui/core';
import { Clear, Done } from '@material-ui/icons';
import isBalanced from '../algorithms/balancer/balancer';

function App() {
  const [balancedResponse, setBalanced] = useState('');

  const handleChange = (e) => {
    const textVal = e.target.value;
    setBalanced(isBalanced(textVal));
  };

  return (
    <Container maxWidth="sm">
      <Box my={4}>
        <Typography variant="h4" component="h1" gutterBottom>
          Balancing Act!
        </Typography>
        <Typography variant="body1" component="section" gutterBottom>
          <TextField id="balancer-input" data-testid="balancer-input" type="text" onChange={handleChange} placeholder="Type some brackets..." />
          {balancedResponse ? <Done data-testid="balanced" /> : <Clear data-testid="not-balanced" />}
        </Typography>
      </Box>
    </Container>
  );
}

export default App;
