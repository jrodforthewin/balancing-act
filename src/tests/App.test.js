import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';

test('App snapshot test', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
