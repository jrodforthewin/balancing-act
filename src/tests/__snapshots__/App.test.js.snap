// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
      />
      <button>
        Send Ref
      </button>
    </form>
    <h3>
      My app
    </h3>
    <div>
      No
    </div>
    Learn react
  </header>
</div>
`;
